R/utils-render.R

```diff
@@ -57,4 +70,185 @@ startRender <- function(data, author, reportName){
   cleanTempSpace()
   
   return(out_file)
+}
+
+
+#' renderCustomFragments
+#' 
+#' Peform the rendering of subfragments for a given report, should return a list of html fragments to 
+#' be available to inject into mustach templates
+#'
+#' @param reportData typed report json, the type should be the name of the report
+#' @return vector of rendered HTML fragments that will be included in the data available to the final report template. The fragments will be
+#' available to the templates as renderedFragments.html
+#' 
+#' @export
+renderCustomFragments <- function(reportData) UseMethod("renderCustomFragments")
```

Comment:
I showed you S3 objects, but this project already has S4 in place, it is probably best not to mix the two.